bonfire/utils.py

```diff
@@ -555,28 +551,29 @@ def hms_to_seconds(s):
 
 
 @lru_cache(maxsize=None)
-def check_hostname(hostname):
+def __check_connection(hostname, port=443, timeout=5):
     """
     Check connection makes sure a connection is available to a given hostname.
 
-    Function is cached so that we only check a hostname once.
+    Function is cached so that we onl
+    y check a hostname once.
     """
     log.debug("checking connection to '%s'", hostname)
 
     try:
-        socket.gethostbyname(hostname)
-    except socket.gaierror:
-        raise ConnectionError(
```

Comment:
The reason that the old code raised a `FatalError` here is that it prints a cleaner message (`"ERROR: <message>"`) and calls `sys.exit(1)` without printing the entire traceback which can make the logs less overwhelming. I feel that single error message is probably going to tell the whole story to the end user and they won't need to see the whole traceback -- so you could `raise FatalError` instead of re-raising TimeoutError on line 571. What do you think?

Reply:
Yeah we can do that - I've updated it to catch the Timeout and raise FatalError instead.